src/app/screens/speedUpTransaction/customFee/index.tsx

```diff
@@ -0,0 +1,129 @@
+import BottomModal from '@components/bottomModal';
+import useWalletSelector from '@hooks/useWalletSelector';
+import { getBtcFiatEquivalent } from '@secretkeylabs/xverse-core';
+import { handleKeyDownFeeRateInput } from '@utils/helper';
+import BigNumber from 'bignumber.js';
+import { useCallback, useEffect, useState } from 'react';
+import { useTranslation } from 'react-i18next';
+import { NumericFormat } from 'react-number-format';
+import {
+  Container,
+  ControlsContainer,
+  FeeContainer,
+  FeeText,
+  InfoContainer,
+  InputContainer,
+  InputField,
+  InputLabel,
+  StyledActionButton,
+  StyledFiatAmountText,
+  StyledInputFeedback,
+  TotalFeeText,
+} from './index.styled';
+
+export default function CustomFee({
+  visible,
+  onClose,
+  onClickApply,
+  calculateTotalFee,
+  feeRate,
+  fee,
+  initialTotalFee,
+  minimumFeeRate,
+  isFeeLoading,
+  error,
+}: {
+  visible: boolean;
+  onClose: () => void;
+  onClickApply: (feeRate: string, fee: string) => void;
+  calculateTotalFee: (feeRate: string) => Promise<number | undefined>;
+  feeRate?: string;
+  fee?: string;
+  minimumFeeRate: string;
+  initialTotalFee: string;
+  isFeeLoading: boolean;
+  error: string;
+}) {
+  const { t } = useTranslation('translation', {
+    keyPrefix: 'TRANSACTION_SETTING',
+  });
+  const { btcFiatRate, fiatCurrency } = useWalletSelector();
+  const [feeRateInput, setFeeRateInput] = useState(feeRate || minimumFeeRate);
+  const [totalFee, setTotalFee] = useState(fee || initialTotalFee);
+
+  const fetchTotalFee = async () => {
+    const response = await calculateTotalFee(feeRateInput);
+
+    if (response) {
+      setTotalFee(response.toString());
+    }
+  };
+
+  useEffect(() => {
+    fetchTotalFee();
+  }, [feeRateInput]);
```

Comment:
```suggestion

  useEffect(() => {
    const fetchTotalFee = async () => {
      const response = await calculateTotalFee(feeRateInput);
  
      if (response) {
        setTotalFee(response.toString());
      }
    };
    fetchTotalFee();
  }, [feeRateInput, calculateTotalFee]);
```

I think if fetchTotalFee is not a function used anywhere but this useEffect, it should go inside to properly manage the trigger conditions

Reply:
When I apply the code you suggested, it starts lagging for some reason 🤔 let's probably keep the original version for now